docs/customruns.md

```diff
@@ -0,0 +1,413 @@
+the `example.dev` API group,  with the version `v1alpha1`.
+
+You can also specify the `name` and optional `namespace` (default is `default`)
+of a custom task resource object previously defined in the cluster.
+
+```yaml
+spec:
+  customRef:
+    apiVersion: example.dev/v1alpha1
+    kind: Example
+    name: my-example
+```
+
+If the `customRef` specifies a name, the custom task controller should look up the
+`Example` resource with that name, and use that object to configure the
+execution.
+
+If the `customRef` does not specify a name, the custom task controller might support
+some default behavior for executing unnamed tasks.
+
+In either case, if the named resource cannot be found, or if unnamed tasks are
+not supported, the custom task controller should update the `CustomRun`'s status to
+indicate the error.
+
+####  Specifying the target Custom Task by embedding its customSpec
+
+To specify the custom task spec, it can be embedded directly into a
+`CustomRun`'s spec as shown below:
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: Run
+metadata:
+  name: embedded-run
+spec:
+  customSpec:
+    apiVersion: example.dev/v1alpha1
+    kind: Example
+    spec:
+      field1: value1
+      field2: value2
+```
+
+This initiates the execution of a `CustomRun` of a custom task of type `Example`, in
+the `example.dev` API group,  with the version `v1alpha1`.
+
+#### Developer guide for custom controllers supporting `customSpec`.
+
+1. A custom controller may or may not support a `Spec`. In cases where it is
+   not supported the custom controller should respond with proper validation
+   error.
+
+2. Validation of the fields of the custom task is delegated to the custom
+   task controller. It is recommended to implement validations as asynchronous
+   (i.e. at reconcile time), rather than part of the webhook. Using a webhook
+   for validation is problematic because, it is not possible to filter custom
+   task resource objects before validation step, as a result each custom task
+   resource has to undergo validation by all the installed custom task
+   controllers.
+
+3. A custom task may have an empty spec, but cannot have an empty
+   `ApiVersion` and `Kind`. Custom task controllers should handle
+   an empty spec, either with a default behaviour, in a case no default
+   behaviour is supported then, appropriate validation error should be
+   updated to the `CustomRun`'s status.
+
+### Cancellation
+
+Implementing cancellation is now `required` instead of `optional`. If the Custom Task implementor does not support cancellation via `.spec.status`, `Pipeline` timeouts and `PipelineRun` cancellation will not work properly.
+
+Pipeline Controller sets the `spec.Status` and `spec.StatusMessage` to signal `CustomRuns` about the `Cancellation`, while `CustomRun` controller **MUST** update its `status.conditions` as following once noticed the change on `spec.Status`.
```

Comment:
We could probably use some background on what custom task controllers are expected to do with status in general. (probably copied from https://github.com/tektoncd/community/blob/main/teps/0002-custom-tasks.md#status-reporting)

Reply:
I'd prefer adding it in this PR-- it's sort of confusing to have instructions about what to do with the status in case of retries or cancelation without understanding the basics of the status in the first place

Reply:
SG, updated!